frontend/packages/console-app/src/components/network-policies/network-policy-selector-preview.tsx

```diff
@@ -0,0 +1,273 @@
+    }
+    // If there is a defined namespace selector, filter pods to remove
+    // those from non-matching namespaces
+    let filteredPods = watchedPods;
+    if (namespaceSelector) {
+      if (watchNsError) {
+        return { error: watchNsError };
+      }
+      filteredPods = filteredPods.filter(
+        (pod) => pod.metadata.namespace && matchedNs.has(pod.metadata.namespace),
+      );
+    }
+    // Group pod TreeViewDataItem by namespace, up to a maximum of maxPreviewedPods entries
+    const podsByNs: { [key: string]: TreeViewDataItem[] } = {};
+    filteredPods.slice(0, maxPreviewPods).forEach((pod) => {
+      const ns = pod?.metadata?.namespace as string;
+      podsByNs[ns] = podsByNs[ns] || [];
+      podsByNs[ns].push({
+        name: pod?.metadata?.name,
+        icon: <ResourceIcon kind={PodModel.kind} />,
+      });
+    });
+    // Then convert the above groups of pod TreeViewDataItems to subchildren of
+    // the namespaces' TreeViewDataItems
+    const podTreeEntries = _.toPairs(podsByNs).map(
+      ([ns, pods]): TreeViewDataItem => ({
+        name: ns,
+        children: pods,
+        defaultExpanded: true,
+        icon: <ResourceIcon kind={NamespaceModel.kind} />,
+      }),
+    );
+    return {
+      pods: podTreeEntries,
+      total: filteredPods.length,
+    };
+  }, [
+    matchedNs,
+    namespaceSelector,
+    selectorError,
+    watchNsError,
+    watchPodError,
+    watchPodLoaded,
+    watchedPods,
+  ]);
+
+  const labelList = _.map(safePodSelector.matchLabels || {}, (value, label) => `${label}=${value}`);
+  const labelBadges = _.map(safePodSelector.matchLabels || {}, (value, label) => (
+    <Label key={label} value={value} color="green">
+      {label}={value}
+    </Label>
+  ));
+  // Filter by labels in the "View all XXX results" link, if needed
+  const podsFilterQuery =
+    preview.total && preview.total > maxPreviewPods && labelList.length > 0
+      ? `?${filterTypeMap.Label}=${encodeURIComponent(
+          labelList.join(labelFilterQueryParamSeparator),
+        )}`
+      : '';
+
+  return preview.error ? (
+    <Alert variant="danger" isInline title={t("public~Can't preview pods")}>
+      <p>{preview.error}</p>
+    </Alert>
+  ) : (
+    <>
+      {watchPodLoaded && preview.pods?.length === 0 && (
+        <div>{t('public~No pods matching the provided labels in the current namespace')}</div>
+      )}
+      {preview.pods && preview.pods.length > 0 && (
+        <>
+          {labelList?.length > 0 ? (
+            <div>
+              {t('public~List of pods matching')} {labelBadges}
+            </div>
+          ) : (
+            <div>{t('public~List of pods')}</div>
+          )}
+          <TreeView data={preview.pods} hasGuides />
+          {preview.total && preview.total > maxPreviewPods && (
+            <>
+              {_.size(safeNsSelector.matchLabels) === 0 ? (
+                <a href={`${resourceListPathFromModel(PodModel, namespace)}${podsFilterQuery}`}>
```

Comment:
I wonder if this link should always open in a new window, wdyt? (@mariomac and @andrew-ronaldson ) I don't imagine anyone while creating a new network policy would like to quit that page to view these search results...
Or alternatively, when asked to view more results, we show all the results within that popup (which then would be scrollable).

Reply:
Do you mean the size of the modal stays the same but it is scrollable to show the list? I have a few other questions if that's alright:

1. What is the benefit the user gets from having this feature in the form?
2. Is there value in seeing the total count of pods or just list them out?
3. Could we in future show the status of the pods? (Example: Failed to connect)

Reply:
Another compromise option could be that, instead of opening the table in a new window, to visualize an embedded table below, like in the original commit.

Reply:
To clarify: first option is to show the popup tree, as now. If the customer clicks into the "View all" link, instead opening a new window, it closes the popup and attaches the preview table below the label selector.

This would also fix an "issue" I noticed with the popup: if I want to update the label selector, the popup is closed while I  change the labels, and then I have to open again the popup. I felt more confortable getting an updated view of the affected pods as long as I typed.

Reply:
> Do you mean the size of the modal stays the same but it is scrollable to show the list? I have a few other questions if that's alright:
> 
>     1. What is the benefit the user gets from having this feature in the form?

IMO, to avoid having to go back and forth between separate views when setting up the labels, as the user tries to make sure the selector targets strictly the desired pods, no more, no less.
If we think showing all pods is to risky (could slow down the page too much), we can keep a limit, but should be much more than 5 IMO
The solution proposed by @mariomac would also work.

> 
>     2. Is there value in seeing the total count of pods or just list them out?

I would say yes, especially if not everything is displayed. But it still has less value that showing them out as the user would still have to list them to make sure everything is correct.

> 
>     3. Could we in future show the status of the pods? (Example: Failed to connect)

Not in the near future, but yes maybe; there's a RFE for checking pods connectivity (RFE-1874)

Reply:
As agreed offline, I have submitted a new PR with the following changes:
- "View all xx results" link now opens in a new tab
- Increased pod limit to 10